Framework/PythonInterface/test/python/plugins/algorithms/FindUBFromScatteringPlaneTest.py

```diff
@@ -0,0 +1,132 @@
+import unittest
+from mantid.simpleapi import SetUB, CreatePeaksWorkspace, FindUBFromScatteringPlane, LoadEmptyInstrument, AddPeakHKL, ClearUB, IndexPeaks
+import numpy as np
+from unittest.mock import patch
+
+
+class FindUBFromScatteringPlaneTest(unittest.TestCase):
+    def setUp(self):
+        # load instrument + empty peakworkspace table
+        self.ws = LoadEmptyInstrument(Filename="SXD_Definition.xml", OutputWorkspace="empty_SXD")
+        axis = self.ws.getAxis(0)
+        axis.setUnit("TOF")
+        self.peaks1 = CreatePeaksWorkspace(InstrumentWorkspace=self.ws, NumberOfPeaks=0, OutputWorkspace="peaks1")
+
+    def assert_vector(self, ref_vector, calculated_vector, tol):
+        self.assertTrue(np.allclose(ref_vector, calculated_vector, atol=tol))
+
+    def index_peaks_helper(self, PeaksWorkspace, tolerance):
+        nindexed, avg_hkl_er, *_ = IndexPeaks(PeaksWorkspace, tolerance, RoundHKLs=False)
+        self.assertLessEqual(avg_hkl_er, 0.5)
+        return nindexed
+
+    def test_find_correct_ub_cubic(self):
+        peaks1 = self.peaks1
+        SetUB(peaks1, u=[1, -0.83, 0], v=[1, 0.83, 0], a=5.4, b=5.4, c=5.4, alpha=90, beta=90, gamma=90)
+        AddPeakHKL(peaks1, [2, 2, 0])
+        ClearUB(peaks1)
+
+        FindUBFromScatteringPlane(
+            Vector1=[1, -1, 0], Vector2=[1, 1, 0], a=5.4, b=5.4, c=5.4, alpha=90, beta=90, gamma=90, PeaksWorkspace="peaks1"
+        )
+        nindexed = self.index_peaks_helper(peaks1, 0.2)
+        self.assertEqual(nindexed, 1)
+
+    def test_find_correct_ub_orthorhombic(self):
+        peaks1 = self.peaks1
+        SetUB(peaks1, u=[0, 0, 1], v=[1, 0, 0])
```

Comment:
As discussed offline missing lattice parameters! Hopefully can reduce some of the tolerances after that!